### PR TITLE
Added input generation (and tests) for WDL 1.0 and CWL

### LINKED ISSUE #3338: Options for Womtool inputs generation
Feature requested by Sam Lee:

An optional flag to use along with the `inputs` command in Womtool to print only required inputs vs all inputs (required & optional)

### PR DESCRIPTION
**Pull request by cjllanwarne**
- Allows any womtool supported language / version to generate an `inputs.json`
- Gives the option to not generate optional inputs
- Adds a whole bunch of test cases which (I'll be honest) I didn't go through super-closely but looked about right
- Closes #3338 (using the `-o false` option

### CODE REVIEW

File: womtool/src/test/resources/inputs/wdl_draft3/singlesample_v1/all.inputs.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "PairedEndSingleSampleWorkflow.ValidateReadGroupSamFile.ignore": "Array[String]?",
+  "PairedEndSingleSampleWorkflow.flowcell_unmapped_bams": "Array[File]",
+  "PairedEndSingleSampleWorkflow.agg_small_disk": "Int",
+  "PairedEndSingleSampleWorkflow.known_indels_sites_VCFs": "Array[File]",
+  "PairedEndSingleSampleWorkflow.wgs_evaluation_interval_list": "File",
+  "PairedEndSingleSampleWorkflow.contamination_sites_vcf": "File",
+  "PairedEndSingleSampleWorkflow.ValidateAggregatedSamFile.ignore": "Array[String]?",
+  "bwa_commandline": "String (default = StringLiteral(bwa mem -K 100000000 -p -v 3 -t 16 -Y $bash_ref_fasta))",
```
Comment (cjllanwarne) [inline womtool/src/test/resources/inputs/wdl_draft3/singlesample_v1/all.inputs.json]: Once #3601 merges, I should probably use the WdlWriter to override WDL 1.0's expression display string

File: womtool/src/test/resources/inputs/wdl_draft3/nested_imports_with_input/all.inputs.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "nested_imports_with_input.os": "WomCompositeType {\n is -> WomCompositeType {\n i -> Int \n} \n}"
```
Comment (cjllanwarne) [inline womtool/src/test/resources/inputs/wdl_draft3/nested_imports_with_input/all.inputs.json]: The UX here could probably be better

File: womtool/src/test/scala/womtool/WomtoolInputsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  private def inputSet(json: String): Set[String] = json.lines.toSet[String].map(_.stripSuffix(","))
+
+  private def expectedJson(versionDirectory: File, caseName: String, jsonName: String): String = {
+    File(mustExist(versionDirectory.path.resolve(caseName).resolve(jsonName).toFile).getAbsolutePath).contentAsString
+  }
+
+  private def mustExist(file: java.io.File): java.io.File = if (file.exists) file else fail(s"No such file: ${file.getAbsolutePath}")
+//  private def ifExists(file: java.io.File): Option[java.io.File] = if (file.exists) Option(file) else None
```
Comment (danbills) [inline womtool/src/test/scala/womtool/WomtoolInputsSpec.scala]: remove?

File: womtool/src/main/scala/womtool/inputs/Inputs.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  private def womTypeToJson(womType: WomType, default: Option[WomExpression]): JsValue = womType match {
+    case WomCompositeType(typeMap) => JsObject(typeMap.map {
+      case (name, wt) => name -> womTypeToJson(wt, None)
+    })
+    case _ =>
+      val defaultString = default.map(d => s"default = ${d.sourceString}").toList
+      val optionalString = if (womType.isInstanceOf[WomOptionalType] || default.isDefined) List("optional") else List.empty
+
+      val suffixStrings = optionalString ++ defaultString
```
Comment (aednichols) [inline womtool/src/main/scala/womtool/inputs/Inputs.scala]: [ToL] Is there a particular reason to make these `List[String]` instead of `String`?

Comment (cjllanwarne) [reply womtool/src/main/scala/womtool/inputs/Inputs.scala]: I agree this was a bit confusing - I've just laid it out as a case statement so make it clearer what's intended here

**Pull request merged**